File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    phone_book.sort()
    for i in range(1, len(phone_book)):
        L = len(phone_book[i-1])
        if phone_book[i-1][:L] == phone_book[i][:L]:
            return False
    return answer

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    bool r=true;
    sort(phone_book.begin(),phone_book.end(),less<string>());
    for(int i=1;i<phone_book.size();i++) {
        if(phone_book[i-1]==phone_book[i].substr(0,phone_book[i-1].size())) {
            r=false;
            break;
        }
    }
    return r;
}